openhcl/virt_mshv_vtl/src/processor/mshv/tlb_lock.rs

```diff
@@ -47,17 +47,30 @@ impl UhProcessor<'_, HypervisorBacked> {
         self.vtls_tlb_locked.set(requesting_vtl, target_vtl, true);
     }
 
+    /// Mark the TLB of the target VTL on the current VP as locked without
+    /// informing the hypervisor. Only should be used when the hypervisor
+    /// is expected to have already locked the TLB.
+    pub(crate) fn mark_tlb_locked(&mut self, requesting_vtl: Vtl, target_vtl: GuestVtl) {
+        debug_assert_eq!(requesting_vtl, Vtl::Vtl2);
+        debug_assert!(self.is_tlb_locked_in_hypervisor(target_vtl));
+        self.vtls_tlb_locked.set(requesting_vtl, target_vtl, true);
+    }
+
     /// Check the status of the TLB lock of the target VTL on the current VP.
-    pub(crate) fn is_tlb_locked(&mut self, requesting_vtl: Vtl, target_vtl: GuestVtl) -> bool {
+    #[cfg(debug_assertions)]
+    pub(crate) fn is_tlb_locked(&self, requesting_vtl: Vtl, target_vtl: GuestVtl) -> bool {
         debug_assert_eq!(requesting_vtl, Vtl::Vtl2);
-        let local_status = self.vtls_tlb_locked.get(requesting_vtl, target_vtl);
+        let locally_locked = self.vtls_tlb_locked.get(requesting_vtl, target_vtl);
         // The hypervisor may lock the TLB without us knowing, but the inverse should never happen.
-        if local_status {
-            debug_assert!(self.is_tlb_locked_in_hypervisor(target_vtl))
-        };
-        local_status
+        if locally_locked {
+            debug_assert!(self.is_tlb_locked_in_hypervisor(target_vtl));
+            true
+        } else {
+            self.is_tlb_locked_in_hypervisor(target_vtl)
```

Comment:
Now that we've patched up the places where we were failing to mark, should we remove this else case?